### ISSUE TITLE
Cancel button leads to SKErrorDomain with code 2

### ISSUE THREAD

**Issue by AllDmeat** [bizz84/SwiftyStoreKit#122]
### Platform

- [x] iOS
- [ ] macOS
- [ ] tvOS

### In app purchase type

- [x] Consumable
- [ ] Non-consumable 
- [ ] Auto-Renewable Subscription
- [ ] Non-Renewing Subscription

### Environment

- [x] Sandbox
- [ ] Production

### Version
0.6.1

### Related issues
couldn't find any

### Issue summary
I'm running debug on my iPhone with iOS 10.2. 

When I atomically call
`purchaseProduct(_ productId: String, atomically: Bool = true, applicationUsername: String = "", completion: @escaping ( PurchaseResult) -> ())`
I'm receiving alert with password confirmation. If I press Cancel I receive error
`failed(Error Domain=SKErrorDomain Code=2 "Cannot connect to iTunes Store" UserInfo={NSLocalizedDescription=Cannot connect to iTunes Store})`

### What did you expect to happen
Some kind of error saying that user canceled purchase by himself.

### What happened instead
I'm receiving error saying that there was connection error.

**Comment by bizz84** [comment]
I have also observed this - this is the error that is returned by `StoreKit` when the user cancels the purchase.

It seems like a generic error so I'm not sure if I can assume that it is only returned in this case and map it to a `.userCanceled` error case.

**Comment by RajChanchal** [comment]
@bizz84 
I needed to check if user himself cancels the alert asking for user's confirmation before buying. This is how I was doing:
```
- (void)failedTransaction:(SKPaymentTransaction *)transaction
{
    if (transaction.error.code != SKErrorPaymentCancelled)
    {
        // error!
          NSLog(@"Something went Wrong!");
        [self finishTransaction:transaction wasSuccessful:NO];
          NSLog(@"transaction error :%@", transaction.error.localizedDescription);
    }
    else
    {
          NSLog(@"Cancelled");
        // this is fine, the user just cancelled
        [[SKPaymentQueue defaultQueue] finishTransaction:transaction];
    }
}
```
Can you please add .userCanceled error case?

**Comment by bizz84** [comment]
Hi @RajChanchal,

At the moment you can do this to detect if the payment was cancelled:

```swift
SwiftyStoreKit.purchaseProduct("your.product.id") { result in
  switch result {
  case .success(let productId):
    print("success")
  case .error(let error): 
    switch error {
    case .failed(let error):
      let skError = SKError(_nsError: error as NSError)
      if skError.code == .paymentCancelled {
        print("Canceled")
      }
      else {
        print("error: \(error)")
      }
    case .invalidProductId(let productId):
      print("\(productId) is not a valid product identifier")
    case .paymentNotAllowed:
      print("You are not allowed to make payments")
    }
  }
}
```

I'm planning to review the error reporting flows so that it's easier to discriminate between various error cases. I may change the `PurchaseResult` type to return `SKError` directly like so:

```swift
public enum PurchaseResult {
    case success(product: Product)
    case error(error: SKError)
}
```
With this, it would then be possible to check all the possible codes as documented here:

https://developer.apple.com/reference/storekit/skerror.code

Who's in favour of this change?

**Comment by bizz84** [comment]
This has now been fixed in version [0.8.0](https://github.com/bizz84/SwiftyStoreKit/releases/tag/0.8.0). Please update.
